Add Authzed Dedicated integration (ECOINT-60) (#2628)
DataDog/integrations-extras: authzed_dedicated/README.md

Reviewer: Note for eng - manually add ## Data Collection section for metrics

# AuthZed Dedicated

## Overview

[Authzed Dedicated][1] is an open-core, [Google Zanzibar][2]-inspired database system for creating and managing security-critical application permissions.

Developers define a schema that models their permissions requirements. They then use any of the official or community-maintained client libraries to apply the schema and insert data into the database. They can query this data to efficiently check permissions within their applications.

Authzed Dedicated metrics allow developers and SREs to monitor their deployments, including request latency, cache metrics (such as size and hit/miss rates), and datastore connection and query performance. These metrics help diagnose performance issues and fine-tune the performance of their SpiceDB clusters.

Sending these metrics to Datadog enables users to leverage their existing observability stack and correlate Authzed Dedicated metrics with other system events.

## Setup

The Datadog integration is available in the Authzed Dashboard under the "Settings" tab on a Permission System.

1.  Go to the dashboard homepage.
2.  Select a Permission System for which to submit metrics.

> 2.  Click on the "Settings" tab.

Reviewer 2: ```suggestion
2.  Click on the **Settings** tab.
```

> 3.  Scroll down to the "Datadog Metrics" block of the settings UI.

Reviewer 2: ```suggestion
3.  Scroll down to the **Datadog Metrics** block of the settings UI.
```

4.  Enter your Datadog account **API key**.

> 5.  Enter your [Datadog Site][3] if different from the default.

Reviewer 2: ```suggestion
5.  Enter your [Datadog site][3] if different from the default.
```

6.  Click **Save**.

You should see metrics start to flow to Datadog shortly thereafter. If you don't, contact [our support][4].

## Uninstallation

> The Datadog integration is available in the Authzed Dashboard under the "Settings" tab on a Permission System.

Reviewer 2: ```suggestion
The Datadog integration is available in the Authzed Dashboard under the **Settings** tab on a Permission System.
```

1.  Go to the dashboard homepage.
2.  Select a Permission System for which to submit metrics.

> 2.  Click on the "Settings" tab.

Reviewer 2: ```suggestion
2.  Click on the **Settings** tab.
```

> 3.  Scroll down to the "Datadog Metrics" block of the settings UI.

Reviewer 2: ```suggestion
3.  Scroll down to the **Datadog Metrics** block of the settings UI.
```

> 4.  Click "Remove."

Reviewer 2: ```suggestion
4.  Click **Remove**.
```

> This will disable the Datadog integration in your Authzed Dedicated cluster. Please note that this could take several minutes.

Reviewer 2: ```suggestion
This disables the Datadog integration in your Authzed Dedicated cluster. Note that this could take several minutes.
```
